RebelAllianceBank/Classes/Customer.cs

```diff
@@ -71,33 +72,107 @@ public void CreateAccount()
-                    case 4:
-                        createAccount = true;
-                        break;
-                    default:
-                        Console.WriteLine("Fel inmatning, inget konto har skapats.");
-                        Console.ReadKey();
-                        Console.Clear();
-                        createAccount = false;
-                        break;
+                    Console.WriteLine($"{account.AccountName} (Saldo: {account.Balance:N2})");
                 }
-            } while (createAccount == false);
+                string currentUserAccountName = Console.ReadLine();
+                currentUserAccount = thecurrentUser.BankAccounts.FirstOrDefault(account => account.AccountName.Equals(currentUserAccountName, StringComparison.OrdinalIgnoreCase));
+                if (currentUserAccount == null)
+                {
+                    Console.WriteLine("Ogiltligt kontonamn.");
+                }
+            }
+            while (otherUser == null)
+            {
+                Console.WriteLine("Vilken användare vill du föra över till? (ange personnummer)");
+                string otherUserPersonalNum = Console.ReadLine();
+                otherUser = users.OfType<Customer>().FirstOrDefault(user => user.PersonalNum == otherUserPersonalNum);
+
+                if (otherUser == null)
+                {
+                    Console.WriteLine("Användaren hittades inte. Tryck enter för att försöka igen.");
+                    while (Console.ReadKey(true).Key != ConsoleKey.Enter) { }
+                    Console.Clear();
+                }
+            }
+            while (otherAccount == null)
+            {
+                Console.WriteLine("Skriv in namnet på kontot du vill föra över till:");
+                foreach (var account in otherUser.BankAccounts)
+                {
+                    Console.WriteLine($"{account.AccountName} (Saldo: {account.Balance:N2})");
+                }
+                string otherAccountName = Console.ReadLine();
+                otherAccount = otherUser.BankAccounts.FirstOrDefault(account => account.AccountName.Equals(otherAccountName, StringComparison.OrdinalIgnoreCase));
+                if (otherAccount == null)
+                {
+                    Console.WriteLine("Ogiltligt kontonamn. Försök igen.");
+                }
+            }
+            decimal amount;
+            while (true)
+            {
+                Console.WriteLine("Hur mycket vill du föra över?");
+                if (decimal.TryParse(Console.ReadLine(), out amount) && amount > 0 && amount <= currentUserAccount.Balance)
+                {
+                    break;
+                }
+                else
+                {
+                    Console.WriteLine("Felaktigt belopp. Det måste vara positivt och inte större än saldot på avsändarkontot.");
+                }
+            }
+            currentUserAccount.Balance -= amount;
+            otherAccount.Balance += amount;
```

Comment:
Should there maybe be a check here to see if the currency of the accounts is the same, and if not maybe convert it properly?

Reply:
Added a comment where method should be placed when done to check currencies.